system/View/View.php

```diff
@@ -219,10 +218,12 @@ public function render(string $view, array $options = null, bool $saveData = nul
 		// Save current vars
 		$renderVars = $this->renderVars;
 
-		ob_start();
-		include $this->renderVars['file']; // PHP will be processed
-		$output = ob_get_contents();
-		@ob_end_clean();
+		$output = (function (string $view, array $data): string {
+			extract($data);
+			ob_start();
+			include $view;
```

Comment:
if `$this` not used  in views then function must be `static` + need add prefix to parameter names to keep their values, in other case need use `$this->renderVars['file']`, `$this->tempData` instead pass as arguments.

Reply:
to test:
~~~php
$this->tempData = ['view' => $this];
~~~